src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -761,10 +740,26 @@ public static BigInteger DivRem(BigInteger dividend, BigInteger divisor, out Big
             if (trivialDivisor)
             {
                 uint rest;
-                uint[] bits = BigIntegerCalculator.Divide(dividend._bits, NumericsHelpers.Abs(divisor._sign), out rest);
 
-                remainder = dividend._sign < 0 ? -1 * rest : rest;
-                return new BigInteger(bits, (dividend._sign < 0) ^ (divisor._sign < 0));
+                uint[]? bitsFromPool = null;
+                int size = dividend._bits.Length;
+                Span<uint> quotient = size <= StackAllocThreshold ?
+                                  stackalloc uint[size]
+                                  : (bitsFromPool = ArrayPool<uint>.Shared.Rent(size)).AsSpan(0, size);
+
+                try
+                {
+                    //may throw DivideByZeroException
+                    BigIntegerCalculator.Divide(dividend._bits, NumericsHelpers.Abs(divisor._sign), quotient, out rest);
+
+                    remainder = dividend._sign < 0 ? -1 * rest : rest;
+                    return new BigInteger(quotient, (dividend._sign < 0) ^ (divisor._sign < 0));
+                }
+                finally
```

Comment:
TBH I'm not sure whether `try-finally` is needed here. I would go with it, but I remember some discussions that say it's not needed. Maybe some else can shed some light into this.

Reply:
`BigIntegerCalculator.Divide` may throw `DivideByZeroException` and buffer should be returned to the pool. Inside of `Divide` method you can find division operator which is the source of the exception. Without finally, rented buffer may leak.